### PR TITLE
async JSON parsing and stringifying

### PR DESCRIPTION
**Pull request by izakp**
Looking into what operations get processed on the event loop versus the worker pool - [this article](https://nodejs.org/en/docs/guides/dont-block-the-event-loop/) pointed out that the native `JSON` module is synchronous.  Given how many cache queries and API responses we are serializing/deserializing in a single request, this may well be a source of latency on the event loop.

Depends on https://github.com/artsy/metaphysics/pull/1107 to measure the event loop latency before and after this is deployed

cc @cavvia

### CODE REVIEW

**Comment by alloy** [comment]
Nice 👌 Feel free to merge when you think is a good time.

**Comment by mbilokonsky** [comment]
Ah, I see the latency measurement will allow us to time it. LGTM I'll merge on green!

File: src/lib/cache.js [hunk 1]
```diff
@@ -21,12 +22,14 @@ const VerboseEvents = ["issue", "failure", "reconnecting", "reconnect", "remove"
...
 const deflateP = (dataz) => {
   return new Promise((resolve, reject) =>
-    zlib.deflate(JSON.stringify(dataz), (er, deflatedData) => {
-      if (er) {
-        error(er)
-      } else {
-        resolve(deflatedData)
-      }
+    yj.stringifyAsync(dataz, (err, data) => {
+      zlib.deflate(JSON.stringify(dataz), (er, deflatedData) => {
```
Comment (acjay) [inline src/lib/cache.js]: If we are actually CPU-bound, is it possible that compression could make things worse?

Comment (acjay) [reply src/lib/cache.js]: Hmm, I see from the article in the PR description that `zlib` runs on a worker thread. So maybe not an issue from an event loop blocking perspective, but still, would add load if our worker threads are also CPU-bound.

Comment (mbilokonsky) [reply src/lib/cache.js]: That load has to happen somewhere, though, right? Moving it to workers can only help, I think?

Comment (acjay) [reply src/lib/cache.js]: Sorry, this wasn't pertaining to using `yj`, but to the recently introduced compression.

Comment (mbilokonsky) [reply src/lib/cache.js]: ooooh, I see what you mean. That's a good point, if we're CPU bound compression would make it worse not better!

**Pull request merged**